docs: fix minValues terminology and clarify capacity-type behavior

moko-poi opened the pull request:
## Description

Fixes terminology error and clarifies capacity-type behavior in NodePools minValues documentation across all versions.

### Issues Fixed
This PR addresses two documentation inaccuracies identified in kubernetes-sigs/karpenter#2435:

1. **Terminology Error**: The documentation incorrectly used "instance families" when referring to [c,m,r], which are actually "instance categories"
   - Instance categories: c, m, r (compute, memory, general purpose)
   - Instance families: c5, m5, r4, c5d, m5d, etc.

2. **Misleading capacity-type behavior**: The text claimed "will use spot instance type for all provisioned instances" but the YAML example didn't specify `karpenter.sh/capacity-type`, meaning both Spot and On-Demand instances could be used.

### Changes Made
- Corrected "instance families" to "instance categories" for [c,m,r]
- Removed misleading claim about Spot-only usage
- Added clarification note explaining that without capacity-type specification, both Spot and On-Demand instances are considered (with Spot preferred due to lower cost)
- Included example showing how to explicitly configure Spot-only instances

### Documentation Versions Affected
- `docs/` (latest)
- `preview/`
- `v1.8/`
- `v1.7/`
- `v1.6/`
- `v1.0/`

Fixes #8800  <!-- replace with aws/karpenter-provider-aws issue number -->
Related: kubernetes-sigs/karpenter#2435

## How was this change tested?

- [x] Reviewed documentation changes for accuracy against implementation
- [x] Verified terminology matches codebase definitions (`karpenter.k8s.aws/instance-category` vs `karpenter.k8s.aws/instance-family`)
- [x] Confirmed all 6 documentation versions have consistent updates
- [x] Validated Hugo alert syntax

## Does this change impact docs?
- [x] Yes, PR includes docs updates
- [ ] Yes, issue opened: #
- [ ] No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

DerekFrank commented on website/content/en/docs/concepts/nodepools.md:
```diff
@@ -301,8 +301,18 @@ There is currently a limit of 100 on the total number of requirements on both th
+For example, the below spec enforces `minValues` to various keys where it is defined:
+at least 2 unique instance categories from [c,m,r], 5 unique instance families [eg: "m5","m5d","r4","c5","c5d","c4" etc], and 10 unique instance types [eg: "c5.2xlarge","c4.xlarge" etc] are required for scheduling the pods.
+
+{{% alert title="Note" color="primary" %}}
```
We could remove this note and just not bother with capacity type. I think this example is trying to explain min values, another example should explain capacity type

DerekFrank reviewed (changes requested):
Thanks for the PR! You don't need to make changes to /docs/ that will get auto updated after merge.

moko-poi committed:
docs: fix minValues example terminology and capacity-type behavior

Fixes terminology error and clarifies capacity-type behavior in NodePools
minValues documentation across all versions.

Changes:
- Correct "instance families" to "instance categories" for [c,m,r]
  (c, m, r are categories, not families like c5, m5, r4)
- Remove misleading claim "will use spot instance type for all provisioned instances"
- Add clarification that without capacity-type specification, both Spot and
  On-Demand instances are considered
- Include example showing how to explicitly specify Spot-only instances

Affects documentation versions:
- preview/
- v1.8/
- v1.7/
- v1.6/
- v1.0/

Related: kubernetes-sigs/karpenter#2435

moko-poi committed:
docs: remove capacity-type note per reviewer feedback

Remove the capacity-type clarification note to keep the example
focused on explaining minValues behavior only, as suggested by reviewer.

The terminology correction (instance families -> instance categories)
is maintained.

engedaam committed:
Merge branch 'main' into docs/fix-minvalues-terminology-and-capacity-type

Sarthug99 commented:
/assign

ryan-mist commented:
Closing this one out as its captured in https://github.com/aws/karpenter-provider-aws/pull/9500
